packages/adapters/bitcoin/src/connectors/UnisatConnector.ts

```diff
@@ -0,0 +1,260 @@
+/* eslint-disable @typescript-eslint/no-unused-vars, @typescript-eslint/require-await */
+import { type CaipNetwork } from '@reown/appkit-common'
+import { CoreHelperUtil, type RequestArguments } from '@reown/appkit-controllers'
+import { bitcoin, bitcoinTestnet } from '@reown/appkit/networks'
+
+import { MethodNotSupportedError } from '../errors/MethodNotSupportedError.js'
+import type { BitcoinConnector } from '../utils/BitcoinConnector.js'
+import { AddressPurpose } from '../utils/BitcoinConnector.js'
+import { ProviderEventEmitter } from '../utils/ProviderEventEmitter.js'
+import { UnitsUtil } from '../utils/UnitsUtil.js'
+
+type WalletId = 'unisat' | 'bitget' | 'binancew3w'
+
+export class UnisatConnector extends ProviderEventEmitter implements BitcoinConnector {
+  public id: WalletId
+  public name
+  public readonly chain = 'bip122'
+  public readonly type = 'ANNOUNCED'
+  public readonly imageUrl: string
+
+  public readonly provider = this
+
+  private readonly wallet: UnisatConnector.Wallet
+  private readonly requestedChains: CaipNetwork[] = []
+  private readonly getActiveNetwork: () => CaipNetwork | undefined
+
+  constructor({
+    id,
+    name,
+    wallet,
+    requestedChains,
+    getActiveNetwork,
+    imageUrl
+  }: UnisatConnector.ConstructorParams) {
+    super()
+    this.id = id
+    this.name = name
+    this.wallet = wallet
+    this.requestedChains = requestedChains
+    this.getActiveNetwork = getActiveNetwork
+    this.imageUrl = imageUrl
+  }
+
+  public get chains() {
+    return this.requestedChains.filter(chain => chain.caipNetworkId === bitcoin.caipNetworkId)
+  }
```

Comment:
Is using `bitcoin` only bc these wallet supports only BTC mainnet? If they support others this would better to do:

```suggestion
 public get chains() {
    return this.requestedChains.filter(chain => chain.chainNamespace === ConstantsUtil.CHAIN.BITCOIN)
  }```

Reply:
`ConstantsUtil` from `@reown/appkit-common`